src/main/java/com/google/sps/servlets/AuthenticateServlet.java

```diff
@@ -44,8 +46,38 @@ public void doGet(HttpServletRequest request, HttpServletResponse  response) thr
     response.setContentType("application/json;");
     Gson gson = new Gson();
 
-    GivrUser user = GivrUser.getLoggedInUser();
+    GivrUser user = GivrUser.getCurrentLoggedInUser();
 
+    Map<String, String> propertyNamesAndValuesToUpdate = new HashMap<String, String>();
+
+    boolean doesUserIdExistInDatastore = GivrUser.checkIfUserWithPropertyExists("userId", user.getUserId());
+    if (!doesUserIdExistInDatastore) {
+      boolean doesUserEmailExistInDatastore = GivrUser.checkIfUserWithPropertyExists("userEmail", user.getUserEmail());
+      if (doesUserEmailExistInDatastore) {
+        // User is either a Moderator or Maintainer, that has not logged in.
+        boolean isMaintainer = user.isMaintainer();
+        if (!isMaintainer) { // User is a Moderator.
+
+          // TODO: Update organization.
```

Comment:
I expected a return here since you don't want to add an entry if they are not a maintainer, right?

Reply:
We want to add an entry if the User is a Maintainer or a Moderator, and if we return here, then we wouldn't be able to add Moderators into our Datastore!